File: Laba_8(c++)/Laba_8(c++)/Matrix.cpp
#include "Matrix.h"


void Matrix::Init()
{

	for (int i = 0; i < n; i++)
	{
		for (int j = 0; j < n; j++)
		{
			M[i][j] = rand() % 100;
		}
	}
}

void Matrix::Output()
{
	for (int i = 0; i < n; i++)
	{
		for (int j = 0; j < n; j++)
		{
			cout << setw(4) << M[i][j];
		}
		cout << endl;
	}
	cout << endl;
}

void Matrix::Summarise()
{
	int sum = 0;
	for (int i = 0; i < n; i++)
	{
		for (int j = 0; j < n; j++)
		{
			sum += M[i][j];
		}
	}
	cout << sum;
}

void Matrix::Summ()
{
	int sum = 0;
	for (int i = 0; i < n; i++)
	{
		for (int j = 0; j < n; j++)
		{
			sum += M[i][j];
		}
	}
	cout << sum;
}

File: Laba_8/Matrix.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Laba_8
{

    class Matrix
    {

        public Random rnd = new Random(); //генерація рандомних значень
        public int sum = 0; //сумма
        public int count = 0; //сумма
        public int rand;
        public int[,] M; //двовимірний масив
        public int n { set; get; } //розмірність масиву

        public Matrix(int n) //конструктор з параметром
        {
            this.n = n;
            M = new int[n, n]; //надання матриці розміру
            Init();
        }



        public void Init() //генерація матриці
        {
            for (int i = 0; i < n; i++)
            {
                for (int j = 0; j < n; j++)
                {
                    rand = rnd.Next(0, 100);
                    M[i, j] = rand;
                }
            }
        }

        public void Output() //вивід матриці
        {
            for (int i = 0; i < n; i++)
            {
                for (int j = 0; j < n; j++)
                {
                    Console.Write(M[i, j] + "\t");
                }
                Console.WriteLine();
            }

        }

        public void Summarise(int[,] arr) //метод знаходження суми елементів
        {
            for (int i = 0; i < 4; i++)
            {
                for (int j = 0; j < 4; j++)
                {
                    sum += arr[i, j];
                }
            }
            Console.WriteLine(sum);
        }

        public void Summ() //віртуальний метод знаходження суми елементів
        {
            for (int i = 0; i < n; i++)
            {
                for (int j = 0; j < n; j++)
                {
                    count += M[i, j];
                }
            }
            Console.WriteLine(count);
        }

    }
}
